contrib/codeanalysis/src/python/pants/contrib/codeanalysis/tasks/index_java.py

```diff
@@ -74,5 +81,22 @@ def entries_file(_vt):
           raise TaskError('java {main} ... exited non-zero ({result})'.format(
             main=self._KYTHE_INDEXER_MAIN, result=result))
 
+    dist = self.get_options().dist
     for vt in invalidation_check.all_vts:
```

Comment:
It's generally a good idea to keep `invalidated` blocks small and readable... breaking the packaging/compression out to another method would help with that.

Reply:
Note that this is outside the `invalidated` block, after populating the products. I figured the user might nuke `dist/` (we don't forbid it, I think) so we should always repopulate it if asked to.

Reply:
But it's still a good idea to break some of this stuff out into helper methods. Done.